libc/src/__support/FPUtil/FPBits.h

```diff
@@ -107,6 +114,14 @@ template <FPType fp_type> struct FPBitsCommon : private FPProperties<fp_type> {
   LIBC_INLINE constexpr bool is_zero() const {
     return (bits & EXP_SIG_MASK) == 0;
   }
+
+  LIBC_INLINE constexpr bool is_inf() const {
+    if constexpr (UP::ENCODING == FPEncoding::X86_ExtendedPrecision) {
```

Comment:
It feels weird to have something specific here. I think we should have the same code for all types, which can simply be `return (bits & EXP_SIG_MASK) == (EXP_MASK | EXPLICIT_BIT_MASK);`. Then `FloatProperties<>::EXPLICIT_BIT_MASK` could be zero except for fp80. Or we can add a `bool REPRESENTS_LEADING_DIGIT`

Reply:
So as discussed off line, let's keep the specialization for `X86_Binary80` to prevent conflating the two implementations. Let's first submit patches up to and including f1b767dcdf453dd5b1beffbfc39c0ff8a5366916 and continue the review process on a different patch.